include/marisa/agent.h

```diff
@@ -36,6 +46,16 @@ class Agent {
     return *state_;
   }
 
+#if __cplusplus >= 201703L
+  void set_key(std::string_view str) {
+    MARISA_DEBUG_IF(str.length() > MARISA_UINT32_MAX, MARISA_SIZE_ERROR);
+    key_.set_str(str);
```

Comment:
Perhaps `set_key(const char *ptr, std::size_t length)` for both C++17 and C++11, and this should call `set_str(str.data(), str.size())`? For symmetry with the `set_str` implementation.

Reply:
Done. This was the first one I implemented and I was "cleverly" trying to avoid extra debug checks. I changed my mind when implementing the other ones.